### PR TITLE
Support custom cuba keys set in particle check templates

### PR DESCRIPTION
**Pull request by itziakos**
This PR streamlines the particle check template to support testing containers that only accept a limited
set of cuba keys.

### CODE REVIEW

**Comment by itziakos** [comment]
@nathanfranklin this is ready for review

File: simphony/testing/utils.py [hunk 1]
```diff
@@ -86,12 +93,18 @@ def create_points():
...
 
 
-def create_bonds(n=5, restrict=None):
+def create_bonds(n=5, restrict=None, particles=None):
     bond_list = []
     for i in xrange(n):
         data = create_data_container(restrict=restrict)
-        data[CUBA.VELOCITY] = i
-        ids = [uuid.uuid4() for x in xrange(n)]
+        # FIXME: This is not right since CUBA.VELOCITY might not be part of the
```
Comment (nathanfranklin) [inline simphony/testing/utils.py]: could the VELOCITY be removed now?

**Comment by nathanfranklin** [comment]
Added one question about the TODO regarding VELOCITY.  Otherwise, :+1:

File: simphony/testing/utils.py (hunk 1 above)
Comment (itziakos) [inline simphony/testing/utils.py]: We need to have a way to differentiate the data containers for the others and at the same time adapt to the fact that the CUBA key that we use should be one of the supported cuba, preferably it should be a predictable value. I have some ideas but I need a little more time to think about it. 

Let me know if you want to give it a go!

**Comment by itziakos** [comment]
@nathanfranklin, I removed the need for the CUBA.VELOCITY. This PR is now ready to review

File: simphony/testing/utils.py [hunk 2]
```diff
@@ -107,34 +114,53 @@ def create_data_container(restrict=None):
...
     -------
     data : DataContainer
 
 
-    """
 
+    """
     if restrict is None:
         restrict = CUBA
     data = {cuba: dummy_cuba_value(cuba) for cuba in restrict}
```
Comment (nathanfranklin) [inline simphony/testing/utils.py]: `constant` is missing from `dummy_cuba_value`

Comment (itziakos) [inline simphony/testing/utils.py]: oops, good catch

Comment (itziakos) [inline simphony/testing/utils.py]: Done

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**